content/en/getting_started/tracing/_index.md

```diff
@@ -20,159 +20,203 @@ further_reading:
+      "Believe you can and you're halfway there. - Theodore Roosevelt",
+      "The future belongs to those who believe in the beauty of their dreams. - Eleanor Roosevelt"
+  ]
+  
+  @app.route('/')
+  def index():
+      quote = random.choice(quotes)+"\n"
+      return quote
+  
+  if __name__ == '__main__':
+      app.run(host='0.0.0.0', port=5050)
+  {{< /code-block >}}
 
-```yaml
-# apm_config:
-##   Whether or not the APM Agent should run
-#   enabled: true
-```
+## Run the application
 
-And in `trace-agent.log`:
+When you set up Datadog APM with Single Step Instrumentation, Datadog automatically instruments your application at runtime.
 
-```bash
-# /var/log/datadog/trace-agent.log:
-2019-03-25 20:33:18 INFO (run.go:136) - trace-agent running on host ubuntu-jammy
-2019-03-25 20:33:18 INFO (api.go:144) - listening for traces at http://localhost:8126
-2019-03-25 20:33:28 INFO (api.go:341) - no data received
-2019-03-25 20:34:18 INFO (service.go:63) - total number of tracked services: 0
-```
+To run `hello.py`:
 
-### Environment name
+1. Create a Python virtual environment in the current directory:
 
-For the best experience, it is recommended to use the environment variable `DD_ENV` to configure `env` through your service's tracer.
+   ```shell
+   python3 -m venv ./venv
+   ```
 
-Additionally, if your tracer has logs injection enabled then the `env` is consistent across traces and logs. Read more about how this works in [Unified Service Tagging][12].
+1. Activate the `venv` virtual environment:
 
-Alternatively, name your environment by updating `datadog.yaml` to set `env` under `apm_config`. To learn more about setting `env` for APM, see the [setting primary tags to scope guide][13].
+   ```shell
+   source ./venv/bin/activate
+   ```
 
-## APM application
+1. Install `pip` and `flask`:
 
-### Installation
+   ```shell
+   sudo apt-get install python3-pip
+   pip install flask
+   ```
 
-Before setting up the application, install `pip` then `flask` and `ddtrace` on your Ubuntu VM:
+1. Run `hello.py`:
 
-```shell
-sudo apt-get install python-pip
-pip install flask
-pip install ddtrace
-```
+   ```shell
+   export DD_SERVICE=hello
+   python3 hello.py
+   ```
 
-### Create
+## Test the application
 
-On the Ubuntu VM, create the application `hello.py` with the following content:
+Test the application to send traces to Datadog:
 
-```python
-from flask import Flask
-app = Flask(__name__)
+1. In a new command prompt, run the following:
 
-@app.route('/')
-def index():
-    return 'hello world'
+   ```shell
+   curl http://0.0.0.0:5050/
+   ```
+1. Confirm that a random quote is returned.
+   ```text
+   Believe you can and you're halfway there. - Theodore Roosevelt
+   ```
 
-if __name__ == '__main__':
-    app.run(host='0.0.0.0', port=5050)
-```
+Each time you run the `curl` command, a new trace is sent to Datadog.
 
-### Run
+## Explore observability data in Datadog
```

Comment:
Let' call it traces. Observability data is very generic

and let' attach a screen shot of traces too.